### PR TITLE
BOAC-2831: Adds appointment search index

### PR DESCRIPTION
**Pull request by lyttam**
https://jira.ets.berkeley.edu/jira/browse/BOAC-2831

### CODE REVIEW

File: boac/models/appointment.py [hunk 1]
```diff
@@ -133,6 +134,7 @@ def create(
             )
         db.session.add(appointment)
         std_commit()
+        cls.refresh_search_index()
```
Comment (johncrossman) [inline boac/models/appointment.py]: @lyttam Are canceled appointments searchable? If not, then we'll need refresh when `cancel()`

Comment (lyttam) [reply boac/models/appointment.py]: @johncrossman I think we'd only need to refresh if `cancel()` changes the details of the appointment.  Cancelled appointments will be searchable unless we add `WHERE canceled_at IS NULL` to the search query.

Comment (lyttam) [reply boac/models/appointment.py]: I just realized we should probably index `cancel_reason` and `cancel_reason_explained`, in which case we will need to refresh the index on cancel.

Comment (johncrossman) [reply boac/models/appointment.py]: @lyttam Would you mind creating a Jira with the details above and question, "what is searchable?" In the meantime, this PR is ready to merge.

**Pull request merged**